src/index.js

```diff
@@ -0,0 +1,113 @@
+// eslint-disable-next-line import/no-internal-modules
+require('@rushstack/eslint-patch/modern-module-resolution');
```

Comment:
From @masone:

> this looks like a workaround for me that I'd like to prevent. can you elaborate on why it's needed?

This actually a workaround, written by Microsoft and have been widely used. Its purpose is giving ability to create a ESLint custom config without  imposing peer dependencies from consumer package:

> This patch is a workaround for a longstanding [ESLint feature request](https://github.com/eslint/eslint/issues/3458) that would allow a shared ESLint config to bring along its own plugins, rather than imposing peer dependencies on every consumer of the config. In a monorepo scenario, this enables your lint setup to be consolidated in a single NPM package.

Read more at: https://www.npmjs.com/package/@rushstack/eslint-patch

Reply:
@masone We have two approaches here:
- either the lint configuration bundles all the needed plugins. This way we can ensure consistency but we need the patch
- we define all the plugins as peer deps. This gives us a bit more flexibility but then it's harder to enforce using the same version of the plugin across different projects

Reply:
Thanks, I can follow on a highl-level. Sounds like you're following an industry standard and that's cool!